synapse/handlers/federation_event.py

```diff
@@ -953,6 +953,41 @@ async def _process_pulled_event(
             )
             return
 
+        # Check if we've already tried to process this event
+        failed_pull_attempt_info = await self._store.get_event_failed_pull_attempt_info(
+            event.room_id, event_id
+        )
+        if failed_pull_attempt_info:
+            # Process previously failed backfill events in the background to not waste
+            # time on something that is bound to fail again.
+            run_as_background_process(
+                "_try_process_pulled_event",
+                self._try_process_pulled_event,
+                origin,
+                event,
+                backfilled,
+            )
```

Comment:
I can't think of a way to test this reliably. We will still try to process the event but just in the background which is timing sensitive of any kind of assertions we would run in a test.

Reply:
Not sure it's worth it though. Mocking and tying the test so closely to the implementation sucks